spec/tasks/mongoid_slug_rake_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper'
+require 'rake'
+
+describe 'mongoid_slug:set' do
+  before :all do
+    load File.expand_path('../../../lib/tasks/mongoid_slug.rake', __FILE__)
+    Rake::Task.define_task(:environment)
+  end
+
+  context 'when models parameter is passed' do
+    let!(:uninitalized_slug1) { UninitalizedSlugFirst.create(name: 'uninitalized_slug1') }
+    let!(:uninitalized_slug2) { UninitalizedSlugSecond.create(name: 'uninitalized_slug2') }
+
+    it 'Goes though all documents of passed models and sets slug if not already set' do
+      class UninitalizedSlugFirst
```

Comment:
I see. Yes, because spec files are loaded in any order and executed in some other you can get in a situation where a subsequent spec depends on the original behavior of the class, and since you reopened and modified it, you now have an unexpected result. Right now this is not a problem in the spec, but in two months someone will update these and we won't know where to begin debugging it ;)

Rewrite these with dynamic classes, ie. `Class.new do ...`?